src/app/modules/timesheet/services/timesheet.service.ts

```diff
@@ -10,7 +10,12 @@ function tokenize(a: any): string {
  ​
 function untokenize(a: string): any {
   if (a) {
-    return JSON.parse(decodeURIComponent(escape(atob(a))));
+    try {
+      return JSON.parse(decodeURIComponent(escape(atob(a))));
+    } catch {
+      alert('Votre URL est érroné');
```

Comment:
Prefer log messages in english.

Reply:
Oh sorry this is for the user. French is fine then. Please change the wording to "Données invalides".